### ISSUE TITLE
How should the hypervisor interpret the physical address returned by OPTEE_SMC_GET_SHM_CONFIG under CFG_VIRTUALIZATION?

### ISSUE THREAD

**Issue by yuvraj1803** [OP-TEE/optee_os#7181]
Assume the guest made an SMC, OPTEE_SMC_GET_SHM_CONFIG which is trapped by the hypervisor. After forwarding this to the secure world (by hooking the VMID in a7), how should the hypervisor work with the physical address provided by OPTEE in a1? Since the guest might not be able to map "its" physical address equivalent (IPA) to the same PA passed by OPTEE. Please correct me if my understanding is wrong.

Regards,
Yuvraj

**Comment by jenswi-linaro** [comment]
With hypervisor, I assume you mean Xen. Upstream Xen doesn't support this for a reason, it doesn't scale well.
A static shared memory carveout needs to be divided into exclusive ranges available for each guest. However, too little shared memory for a guest might lead to sporadic troubles since it may run out of shared memory from time to time.

**Comment by yuvraj1803** [comment]
I see. I should have mentioned, I am playing around with KVM (experimenting with OPTEE virtualization support) actually. After digging in I realized, Linux's OP-TEE driver checks for OPTEE_SMC_SEC_CAP_HAVE_RESERVED_SHM.

I guess a way around this is to return NO to a request from a guest asking for OPTEE_SMC_SEC_CAP_HAVE_RESERVED_SHM.

**Comment by jenswi-linaro** [comment]
Yes, the Xen code may be of some help as a reference. https://github.com/xen-project/xen/blob/master/xen/arch/arm/tee/optee.c

**Comment by yuvraj1803** [comment]
Much helpful. Thanks! @jenswi-linaro
